cl/recap/management/commands/clean_up_docket_judges.py

```diff
@@ -0,0 +1,89 @@
+# !/usr/bin/python
+# -*- coding: utf-8 -*-
+import time
+
+from asgiref.sync import async_to_sync
+from django.db.models import Q
+
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.people_db.lookup_utils import lookup_judge_by_full_name
+from cl.search.models import Docket
+
+
+def find_and_fix_docket_judges(
+    iteration_wait: float, testing_mode: bool
+) -> None:
+    """Find and fix Dockets referred_to and assigned_to judges.
+
+    :param iteration_wait: A float that indicates the time to wait between iterations.
+    :param testing_mode: True if testing mode is enabled to avoid wait between
+    iterations.
+    :return: None
+    """
+
+    logger.info("Finding dockets with judges.")
+    dockets_with_judges = Docket.objects.filter(
+        Q(referred_to__isnull=False) | Q(assigned_to__isnull=False)
+    )
+    total_dockets = dockets_with_judges.count()
+    logger.info(f"Total dockets to process: {total_dockets}")
+
+    fixed_dockets = 0
+    for iteration, d in enumerate(dockets_with_judges.iterator(), start=1):
+        new_referred = (
+            async_to_sync(lookup_judge_by_full_name)(
+                d.referred_to_str, d.court_id, d.date_filed
+            )
+            if d.referred_to_str
+            else None
+        )
+        new_assigned = (
+            async_to_sync(lookup_judge_by_full_name)(
+                d.assigned_to_str, d.court_id, d.date_filed
+            )
+            if d.assigned_to_str
+            else None
```

Comment:
Could you use [lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache) here to avoid zillions of DB lookups? Maybe set a maxsize of, say 30k so that we catch most judge names?